Ensure that s3_bucket versioning is correctly enabled

willthames opened the pull request:
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
s3_bucket

##### ANSIBLE VERSION
```
ansible 2.3.0 (devel a6cb377420) last updated 2017/02/13 10:23:20 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
If versioning is not enabled, then `get_bucket_versioning`
can return an empty dict.

If that happens, the code to enable versioning should still
run!

The logic for suspending versioning was also incorrect, so
have updated that too.

Fixes #20491

Tested with:
```
hacking/test-module -m lib/ansible/modules/cloud/amazon/s3_bucket.py -a "profile=sandbox name=test-2017-02-13-1039 versioning=no"
 hacking/test-module -m lib/ansible/modules/cloud/amazon/s3_bucket.py -a "profile=sandbox name=test-2017-02-13-1039 versioning=yes"
 hacking/test-module -m lib/ansible/modules/cloud/amazon/s3_bucket.py -a "profile=sandbox name=test-2017-02-13-1039 versioning=no"
```

and results now as expected

Commit:
Ensure that s3_bucket versioning is correctly enabled

If versioning is not enabled, then `get_bucket_versioning`
can return an empty dict.

If that happens, the code to enable versioning should still
run!

The logic for suspending versioning was also incorrect, so
have updated that too.

Fixes #20491

ryansb commented on lib/ansible/modules/cloud/amazon/s3_bucket.py:
```diff
@@ -168,22 +168,21 @@ def _create_or_update_bucket(connection, module, location):
+                changed = True
+                versioning_status = bucket.get_versioning_status()
+            except S3ResponseError as e:
+                module.fail_json(msg=e.message)
```
Please add the traceback to this message with `exception=traceback.format_exc()`

willthames replied:
This code isn't actually changing (apart from the change to indentation)

There is no documentation that exceptions should be returned when calling `fail_json`.

I can do this, but it's a surprise that I should need to.

Commit:
Added exception traceback and fixed pep8 issues

Added traceback to exceptions, and updated imports and
other pep8 issues while in the area.

willthames commented:
@ryansb traceback added to exceptions and various pep8 issues resolved

willthames commented:
!needs_revision
